### PR TITLE
[BugFix] Fix NPE when publish version with mv

### LINKED ISSUE #15793: errors while publish version to all backends
### Real behavior (Required)
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.5

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15793

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|4|6|66.67%|[407, 408]|

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java [hunk 1]
```diff
@@ -395,11 +396,18 @@ private void refreshMvIfNecessary(TransactionState transactionState)
...
+                MvId mvId = mvIdIterator.next();
                 Database mvDb = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb(mvId.getDbId());
                 mvDb.readLock();
                 try {
                     MaterializedView materializedView = (MaterializedView) mvDb.getTable(mvId.getId());
+                    if (materializedView == null) {
+                        LOG.warn("materialized view {} does not exists.", mvId.getId());
+                        mvIdIterator.remove();
+                        continue;
+                    }
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java:410]: could you add an UT to cover this logic?

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by gengjun-git — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**